pkg/operator/options/options.go

```diff
@@ -0,0 +1,177 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/settings"
+	coreoptions "github.com/aws/karpenter-core/pkg/operator/options"
+	"github.com/aws/karpenter-core/pkg/utils/env"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func init() {
+	coreoptions.Injectables = append(coreoptions.Injectables, &Options{})
+}
+
+type nodeIdentitiesValue []string
+
+func newNodeIdentitiesValue(val string, p *[]string) *nodeIdentitiesValue {
+	*p = strings.Split(val, ",")
+	return (*nodeIdentitiesValue)(p)
+}
+
+func (s *nodeIdentitiesValue) Set(val string) error {
+	*s = nodeIdentitiesValue(strings.Split(val, ","))
+	return nil
+}
+
+func (s *nodeIdentitiesValue) Get() any { return []string(*s) }
+
+func (s *nodeIdentitiesValue) String() string { return strings.Join(*s, ",") }
+
+type optionsKey struct{}
+
+type Options struct {
+	ClusterName                    string
+	ClusterEndpoint                string // => APIServerName in bootstrap, except needs to be w/o https/port
+	VMMemoryOverheadPercent        float64
+	ClusterID                      string
+	KubeletClientTLSBootstrapToken string   // => TLSBootstrapToken in bootstrap (may need to be per node/nodepool)
+	SSHPublicKey                   string   // ssh.publicKeys.keyData => VM SSH public key // TODO: move to node template?
+	NetworkPlugin                  string   // => NetworkPlugin in bootstrap
+	NetworkPolicy                  string   // => NetworkPolicy in bootstrap
+	NodeIdentities                 []string // => Applied onto each VM
+
+	setFlags map[string]bool
+}
+
+func (o *Options) AddFlags(fs *coreoptions.FlagSet) {
+	fs.StringVar(&o.ClusterName, "cluster-name", env.WithDefaultString("CLUSTER_NAME", ""), "[REQUIRED] The kubernetes cluster name for resource discovery.")
+	fs.StringVar(&o.ClusterEndpoint, "cluster-endpoint", env.WithDefaultString("CLUSTER_ENDPOINT", ""), "[REQUIRED] The external kubernetes cluster endpoint for new nodes to connect with. If not specified, will discover the cluster endpoint using DescribeCluster API.")
```

Comment:
```suggestion
	fs.StringVar(&o.ClusterEndpoint, "cluster-endpoint", env.WithDefaultString("CLUSTER_ENDPOINT", ""), "[REQUIRED] The external kubernetes cluster endpoint for new nodes to connect with.")
```
Has to be specified (and DescribeCluster is an EKS thing :-)

Reply:
Damn this copilot code slipped through my eyes